Context object has no attribute 'target_info'

While implementing `np.cumsum`, I stumble across this issue:

```
E                   numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
E                   Invalid use of Function(<function cumsum at 0x119b3b3a0>) with argument(s) of type(s): ((STRUCT__lPLbK)*)
E                    * parameterized
E                   In definition 0:
E                       All templates rejected with literals.
E                   In definition 1:
E                       All templates rejected without literals.
E                   In definition 2:
E                       TypingError: Failed in nopython mode pipeline (step: nopython frontend)
E                   Invalid use of Function(<class 'rbc.omnisci_array.Array'>) with argument(s) of type(s): (int64, class(int64))
E                    * parameterized
E                   In definition 0:
E                       AttributeError: 'Context' object has no attribute 'target_info'
E                       raised from /Users/guilhermeleobas/Documents/GitHub/rbc/rbc/omnisci_array.py:86
E                   In definition 1:
E                       AttributeError: 'Context' object has no attribute 'target_info'
E                       raised from /Users/guilhermeleobas/Documents/GitHub/rbc/rbc/omnisci_array.py:86
E                   This error is usually caused by passing an argument of a type that is unsupported by the named function.
E                   [1] During: resolving callee type: Function(<class 'rbc.omnisci_array.Array'>)
E                   [2] During: typing of call at /Users/guilhermeleobas/Documents/GitHub/rbc/rbc/omnisci_array.py (202)
E
E
E                   File "rbc/omnisci_array.py", line 202:
E                           def impl(a):
E                               <source elided>
E                               sz = len(a)
E                               out = Array(sz, eltype)
E                               ^
```

I believe the error happens here:
```python
@extending.type_callable(Array)
def type_omnisci_array(context): # <--- this context is not the one on irtools.py:226
    def typer(size, dtype):
        return array_type_converter(context.target_info, dtype).tonumba()
    return typer
```

I guess that Numba creates a new context when analyzing other functions. For instance, in the example below, the test calls `np.cumsum`. To analyze `np.cumsum`, **I think** Numba creates a new typing context.

## Steps to reproduce

#### `np.cumsum` implementation:
```python
@extending.overload(np.cumsum)
def omnisci_np_cumsum(a):
    if isinstance(a, ArrayPointer):
        eltype = a.eltype
        def impl(a):
            sz = len(a)
            out = Array(sz, eltype)
            out[0] = a[0]
            for i in range(1, sz):
                out[i] = out[i-1] + a[i]
            return out
        return impl
```

#### Test
```python
def test_np_cumsum(omnisci):
    omnisci.reset()

    from rbc.omnisci_array import Array
    import numpy as np

    @omnisci('int64[](int64[])')
    def np_cumsum(a):
        return np.cumsum(a)

    query = (
        'select i8, np_cumsum(i8) from {omnisci.table_name};'
        .format(**locals()))
    _, result = omnisci.sql_execute(query)

    r = list(result)[0]
    print(r)
```

--- a/rbc/irtools.py
+++ b/rbc/irtools.py
@@ -227,6 +227,9 @@ def compile_to_LLVM(functions_and_signatures, target: TargetInfo,
         target_context = RemoteCPUContext(typing_context, target)
         # Bring over Array overloads (a hack):
         target_context._defns = target_desc.target_context._defns
+        # Fixes rbc issue 74:
+        target_desc.typing_context.target_info = target
+        target_desc.target_context.target_info = target
 
     typing_context.target_info = target
     target_context.target_info = target

--- a/rbc/omnisci_array.py
+++ b/rbc/omnisci_array.py
@@ -297,6 +297,21 @@ def omnisci_array_mean(x):
         return impl
 
 
+@extending.overload(np.cumsum)
+def omnisci_np_cumsum(a):
+    if isinstance(a, ArrayPointer):
+        eltype = a.eltype
+
+        def impl(a):
+            sz = len(a)
+            out = Array(sz, eltype)
+            out[0] = a[0]
+            for i in range(1, sz):
+                out[i] = out[i-1] + a[i]
+            return out
+        return impl
+
+
 _array_type_match = re.compile(r'\A(.*)\s*[\[]\s*[\]]\Z').match
 
 

--- a/rbc/tests/test_omnisci_array.py
+++ b/rbc/tests/test_omnisci_array.py
@@ -406,3 +406,22 @@ def test_array_constructor_is_null(omnisci):
     _, result = omnisci.sql_execute(query)
 
     assert list(result)[0] == (0,)
+
+
+def test_np_cumsum(omnisci):
+    omnisci.reset()
+
+    import numpy as np
+
+    @omnisci('int64[](int64[])')
+    def np_cumsum(a):
+        return np.cumsum(a)
+
+    query = (
+        'select i8, np_cumsum(i8) from {omnisci.table_name};'
+        .format(**locals()))
+    _, result = omnisci.sql_execute(query)
+
+    inp, out = list(result)[0]
+    expected = np.cumsum(inp)
+    assert np.isclose(expected, out, equal_nan=True).all()

